src/components/EventDescription.js

```diff
@@ -0,0 +1,263 @@
+let settingRegistrationData = false;
+const useStyles = makeStyles(theme => ({
+  layout: {
+    [theme.breakpoints.up("sm")]: {
+      width: 600,
+      margin: "auto"
+    }
+  },
+  paper: {
+    maxWidth: "1274px",
+    padding: "60px 0 80px 95px",
+    position: "relative"
+  },
+
+  title: {
+    display: "inline-block",
+    fontSize: "36px"
+  },
+  description: {
+    margin: "50px 0px 67px 0px"
+  },
+
+  favLogo: {
+    position: "absolute",
+    top: "4.5px",
+    right: "100px",
+    fill: COLOR.BIZTECH_GREEN,
+    fontSize: "32px"
+  },
+  viewLogo: {
+    position: "absolute",
+    right: "150px",
+    top: "6px",
+    fontSize: "32px"
+  },
+  button: {
+    marginLeft: "10px",
+    color: COLOR.WHITE,
+    padding: "6px 12px"
+  },
+  buttonGroup: {
+    position: "absolute",
+    right: "100px"
+  }
+}));
+
+const sendFavouriteData = async (userID, eventID, isFavourite) => {
+  if (settingFavouriteData === true) {
+    return Promise.resolve("in_progress");
+  }
+  settingFavouriteData = true;
+  const bodyData = {
+    eventID: eventID,
+    isFavourite: isFavourite
+  };
+  try {
+    await fetchBackend(`/users/favEvent/${userID}`, "PATCH", bodyData);
+    settingFavouriteData = false;
+    let responesMsg = "";
+    isFavourite ? (responesMsg = "favourite") : (responesMsg = "unfavourite");
+    responesMsg += " succeed";
+    return Promise.resolve(responesMsg);
+  } catch (error) {
+    settingFavouriteData = false;
+    return Promise.reject(error);
+  }
+};
+
+const sendRegistrationData = async (id, eventID, isRegister, isFirstTime) => {
+  if (settingRegistrationData === true) {
+    return Promise.resolve("in_progress");
+  }
+  settingRegistrationData = true;
+  let registrationStatus = "";
+  let method = "";
+  let path = "";
+  let body = {
+    eventID: eventID,
+    registrationStatus: registrationStatus
+  };
+  if (isRegister) {
+    registrationStatus = REGISTRATION_STATUS.REGISTERED;
+  } else {
+    registrationStatus = REGISTRATION_STATUS.CANCELLED;
+  }
+  if (isFirstTime) {
+    body["id"] = id;
+    method = "POST";
+    path = "/registrations";
+  } else {
+    method = "PUT";
+    path = `/registrations/${id}`;
+  }
+  try {
+    await fetchBackend(path, method, body);
+    settingRegistrationData = false;
+    let responesMsg = "";
+    isRegister
+      ? (responesMsg = "registration")
+      : (responesMsg = "unregistration");
+    responesMsg += " succeed";
+    return Promise.resolve(responesMsg);
+  } catch (error) {
+    settingRegistrationData = false;
+    return Promise.reject(error);
+  }
+};
+
+const TransitionUp = props => {
+  return <Slide {...props} direction="up" />;
+};
+
+const EventDescription = ({ user, event, registration, children }) => {
+  const classes = useStyles();
+  const [eventFavStatus, setEventFavStatus] = useState(false);
+  const [eventRegistrationStatus, setEventRegistrationStatus] = useState(false);
+  const [snackOpen, setSnackOpen] = React.useState(false);
```

Comment:
is there a way to set whether or not the snackbar is open without using a state variable?